rhc-ose-ansible/roles/registry/templates/nginx.conf

```diff
@@ -0,0 +1,64 @@
+events {
+    worker_connections  1024;
+}
+
+http {
+
+  upstream docker-registry {
+    server localhost:5000;
+  }
+
+  ## Set a variable to help us decide if we need to add the
+  ## 'Docker-Distribution-Api-Version' header.
+  ## The registry always sets this header.
+  ## In the case of nginx performing auth, the header will be unset
+  ## since nginx is auth-ing before proxying.
+  map $upstream_http_docker_distribution_api_version $docker_distribution_api_version {
+    'registry/2.0' '';
+    default registry/2.0;
+  }
+
+  server {
+    listen 443 ssl;
+    server_name {{template_registry_domain_name}};
+
+    # SSL
+    ssl_certificate /etc/nginx/ssl/server.crt;
+    ssl_certificate_key /etc/nginx/ssl/server.key;
+
+    # Recommendations from https://raymii.org/s/tutorials/Strong_SSL_Security_On_nginx.html
+    ssl_protocols TLSv1.1 TLSv1.2;
+    ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';
+    ssl_prefer_server_ciphers on;
+    ssl_session_cache shared:SSL:10m;
+
+    # disable any limits to avoid HTTP 413 for large image uploads
+    client_max_body_size 0;
+
+    # required to avoid HTTP 411: see Issue #1486 (https://github.com/docker/docker/issues/1486)
+    chunked_transfer_encoding on;
+
+    location /v2/ {
+      # Do not allow connections from docker 1.5 and earlier
+      # docker pre-1.6.0 did not properly set the user agent on ping, catch "Go *" user agents
+      if ($http_user_agent ~ "^(docker\/1\.(3|4|5(?!\.[0-9]-dev))|Go ).*\$" ) {
+        return 404;
+      }
+
+      # To add basic authentication to v2 use auth_basic setting.
+      auth_basic "Registry realm";
+      auth_basic_user_file /etc/nginx/conf.d/nginx.htpasswd;
+
+      ## If $docker_distribution_api_version is empty, the header will not be added.
+      ## See the map directive above where this variable is defined.
+      add_header 'Docker-Distribution-Api-Version' \$docker_distribution_api_version always;
```

Comment:
the `Docker-Distribution-Api-Version` header must be set to `registry/2.0`, otherwise there are auth issues. It was extracted out as a variable in the example docs to support multiple api versions. Since we are only going to support v2, statically setting this is harmless and required as follows: 

    add_header 'Docker-Distribution-Api-Version' 'registry/2.0' always;